### PR TITLE
Build on every push but only publish on tags

### PR DESCRIPTION
**Pull request by adriangb**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -259,15 +260,14 @@ jobs:
...
-      - lint
       - macos
       - windows
       - linux
       - linux-cross
       - musllinux
       - musllinux-cross
       - pypy
-    if: "startsWith(github.ref, 'refs/tags/')"
+    if: "github.ref == 'refs/heads/main' && startsWith(github.ref, 'refs/tags/')"
```
Comment (charliermarsh) [inline .github/workflows/release.yaml:270]: Can this condition be true? If `github.ref` is exactly `'refs/heads/main'`, then how could it start with `'refs/tags/'`?

Comment (adriangb) [reply .github/workflows/release.yaml]: Done!

**Comment by adriangb** [comment]
Now all this is doing is running the workflow on every push (which generates the artifacts as a GitHub artifact tied to the commit) but doesn't publish them on PyPi unless there's a tag. Also fixing a leftover from `lint`. I think you'll be able to get rid of the `maturin_build` and maybe `cargo built` jobs in `ci.yaml` right?

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**